components/Tabs/src/TabBarView/MDCTabBarView.m

```diff
@@ -89,6 +103,63 @@ - (void)setSelectedItem:(UITabBarItem *)selectedItem {
   }
 
   _selectedItem = selectedItem;
+  if (itemIndex == NSNotFound) {
+    return;
+  }
+
+  UIView *itemView = self.itemViews[itemIndex];
+  if ([itemView isKindOfClass:[MDCTabBarViewItemView class]]) {
+    MDCTabBarViewItemView *selectedItemView = (MDCTabBarViewItemView *)itemView;
+    selectedItemView.iconImageView.image = selectedItem.image;
+  }
+}
+
+#pragma mark - Key-Value Observing (KVO)
+
+- (void)addObserversToTabBarItems {
+  for (UITabBarItem *item in self.items) {
+    [item addObserver:self
+           forKeyPath:kImageKeyPath
+              options:NSKeyValueObservingOptionNew
+              context:kKVOContextMDCTabBarView];
+    [item addObserver:self
+           forKeyPath:kTitleKeyPath
+              options:NSKeyValueObservingOptionNew
+              context:kKVOContextMDCTabBarView];
+  }
+}
+
+- (void)removeObserversFromTabBarItems {
+  for (UITabBarItem *item in self.items) {
+    [item removeObserver:self forKeyPath:kImageKeyPath context:kKVOContextMDCTabBarView];
+    [item removeObserver:self forKeyPath:kTitleKeyPath context:kKVOContextMDCTabBarView];
```

Comment:
Did you verify that removing an observer that doesn't exist doesn't cause a crash?  If not, is it possible to check if the observer exists before removing it?

Reply:
There was crash-checking code previously (catching an NSException) but it was removed as a request from @jverkoey.  I think we should crash if we make a mistake rather than double-removing observers.